### ISSUE TITLE
Translations for projects contents (layer names, group names, column names, metadata, etc.)

### ISSUE THREAD

**Issue by gioman** [3liz/lizmap-web-client#453]
This is a suggestion for an improvement, if this is already possible somehow then please disregard this ticket and close.

LM supports web client interface (and backoffice) translations that automatically kick in when the OS/Browser language matches one of them, but seems that there is not yet mechanism to support projects contents that need to be published in different languages, that then need to be cloned/copied and translated within QGIS or LM plugin.

A suggestion (one suggestion, to see if is the one that makes more sense) would be to allow translate layers names/groups/metadata within the .cfg (using LM plugin) of one single QGIS project and allow define a default language.

If there is a content translation for a specific language matching the one being used for the interface, then override the default strings, otherwise use default strings.

I would be interested in contributing the implementation of such or similar mechanism.

Regards

**Comment by pcav** [comment]
Very nice idea, thanks Giovanni. Maybe this could also be implemented upstream? I can think of multilingual desktop projects as well.

**Comment by rldhont** [comment]
Is it the same enhancement as #180 ?

**Comment by gioman** [comment]
@rldhont yes I guess so. Sorry I missed it. 

I guess/agree that ideally this mechanism would be best added in upstream qgis (with translated strings for contents then saved to LM .cfg files) but I think that is especially important for the web publishing aspect of the workflow.

**Comment by gioman** [comment]
see also https://github.com/qgis/QGIS-Enhancement-Proposals/issues/90

**Comment by igeofr** [comment]
+1 
I even have a problem on a project (FR/EN)

**1. For popup I use the tag "br" in a name field**
<img width="434" alt="image" src="https://user-images.githubusercontent.com/7761073/68868197-00e24280-06ef-11ea-998a-780aae0fe6d0.png">

**2. But this does not work for the form being edited**
<img width="335" alt="image" src="https://user-images.githubusercontent.com/7761073/68868109-d4c6c180-06ee-11ea-9e31-c165199280ee.png">

**Comment by gioman** [comment]
@rldhont @Gustry the QEP https://github.com/qgis/QGIS-Enhancement-Proposals/issues/90 has been implemented, is there any possible path to make use of it in LMWC?

if the functionality is supported by QGIS server with a language parameter (which I'm not sure it is) this could be set from the LMWC plugin or dynamically getting the language of the browser(?).

**Comment by Gustry** [comment]
Have a look to https://github.com/3liz/lizmap-javascript-scripts/tree/master/library/misc/translation how to do that in LWC for now.

For qgis/QGIS-Enhancement-Proposals#90, it would need to be managed in QGIS Server, because this is generating a TS file. So it's not possible to use outside of Qt.

**Comment by gioman** [comment]
> Have a look to https://github.com/3liz/lizmap-javascript-scripts/tree/master/library/misc/translation how to do that in LWC for now.

@Gustry thanks! I didn't know about this. One questions: if a translation file is available, it is applied automatically depending on browser language?

**Comment by Gustry** [comment]
> One questions: if a translation file is available, it is applied automatically depending on browser language?

I can't answer. It was a feature made by @RobiFag and @rmarzocchi84 ! ;-)

**Comment by RobiFag** [comment]
> > Have a look to https://github.com/3liz/lizmap-javascript-scripts/tree/master/library/misc/translation how to do that in LWC for now.
> 
> @Gustry thanks! I didn't know about this. One questions: if a translation file is available, it is applied automatically depending on browser language?

Yes, the proper translation file is used depending on browser language but you have to specify the language ID code in the translation.js file (see the comments in the js file)!

**Comment by gioman** [comment]
> Yes, the proper translation file is used depending on browser language but you have to specify the language ID code in the translation.js file (see the comments in the js file)!

@RobiFag thanks!
